howso/typing.amlg

```diff
@@ -0,0 +1,206 @@
+                                type "assoc"
+                                indices
+                                    (assoc
+                                        type {type "string" enum ["rate" "delta"]}
+                                        time_feature "boolean"
+                                        order "number"
+                                        derived_orders "number"
+                                        lags {type "list" values "number"}
+                                        num_lags "number"
+                                        rate_min {type "list" values "number"}
+                                        rate_max {type "list" values "number"}
+                                        delta_min {type "list" values "number"}
+                                        delta_max {type "list" values "number"}
+                                        series_has_terminators "boolean"
+                                        stop_on_terminator "boolean"
+                                        universal "boolean"
+                                    )
+                            )
+                    )
+            )
+        CaseIndices
+            (assoc
+                type "list"
+                values
+                    (assoc
+                        type "list"
+                        values (list "string" "number")
+                        min_size 2
+                        max_size 2
+                    )
+            )
+        Condition
+            (assoc
+                type "assoc"
+                values "any"
+            )
+        Precision (assoc type "string" enum (list "exact" "similar"))
+        FeatureBoundsMap
+            (assoc
+                type "assoc"
+                values
+                    (assoc
+                        type "assoc"
+                        indices
+                            (assoc
+                                min "number"
+                                max "number"
+                                allow_null "boolean"
+                                allowed {type "list" values "any"}
+                            )
+                    )
+            )
+        HyperparameterMap
+            (assoc
+                type "assoc"
+                indices
+                    (assoc
+                        k "number"
+                        p "number"
+                        dt "number"
+                        featureDeviations (assoc type "assoc" values "any" optional (true))
+                        featureDomainAttributes (assoc type "assoc" values "any")
+                        featureWeights (assoc type "assoc" values "number" optional (true))
+                        paramPath (assoc type "list" values "string")
+                        useDeviations "boolean"
+                        allFeatureResidualsCached "boolean"
+                        !featureOrdinalDeviations (assoc type "assoc" values "any")
```

Comment:
Is this correct? Should the key have an exclamation point in it or is it intended to resolve the value of `!featureOrdinalDeviations`?

Reply:
its unfortunately real for now, we should resolve/remove this later though